src/librustc_typeck/diagnostics.rs

```diff
@@ -1250,6 +1250,43 @@ information see the [opt-in builtin traits RFC](https://github.com/rust-lang/
 rfcs/blob/master/text/0019-opt-in-builtin-traits.md).
 "##,
 
+E0195: r##"
+Your method's lifetime parameters do not match the trait declaration.
+Erroneous code example:
+
+```
+trait Trait {
+    fn t<'a,'b:'a>(x: &'a str, y: &'b str);
+}
+
+struct Foo;
+
+impl Trait for Foo {
+    fn t<'a,'b>(x: &'a str, y: &'b str) { // error: lifetime parameters
+                                          //        or bounds on method `t`
+                                          //        do not match the trait
+                                          //        declaration
+    }
+}
+```
+
+Here, the 'b lifetime parameter must live longer than 'a. So please check
```

Comment:
> Here, the 'b lifetime parameter must live longer than 'a. 

I’d argue that you want to comment as generally as possible, not about your exact example case, for example

> The 'b lifetime constraints for `t` implementation does not match the trait declaration. Ensure lifetime declarations match exactly in both trait declaration and implementation.

or something along the lines.